include/KahanAccumulator.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+/**
+ * Kahan summation implementation.
+ *
+ * The algorithm is taken from
+ * https://en.wikipedia.org/wiki/Kahan_summation_algorithm
+ *
+ * @tparam Numeric A numeric type which supports addition and subtraction. The
+ * most sensible values are `double` and `std::complex<double>`.
+ */
+template <typename Numeric>
+class KahanAccumulator {
+ public:
+  Numeric value() const { return sum_; }
```

Comment:
This should apply the final correction, no?

Reply:
See https://github.com/etmc/tmLQCD/blob/master/kahan_summation.h

https://github.com/etmc/tmLQCD/blob/64ac1b3eb88299b5cd28ffc2bde7c31030974815/kahan_summation.h#L41

https://github.com/etmc/tmLQCD/blob/64ac1b3eb88299b5cd28ffc2bde7c31030974815/kahan_summation.h#L49

Reply:
Yes, that makes more sense.